Mathlib/LinearAlgebra/CliffordAlgebra/Prod.lean

```diff
@@ -0,0 +1,169 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.LinearAlgebra.CliffordAlgebra.Grading
+import Mathlib.LinearAlgebra.TensorProduct.Graded.Internal
+import Mathlib.LinearAlgebra.QuadraticForm.Prod
+
+/-!
+# Clifford algebras of a direct sum of two vector spaces
+
+We show that the Clifford algebra of a direct sum is the graded tensor product of the Clifford
+algebras, as `CliffordAlgebra.equivProd`.
+
+## Main definitions:
+
+* `CliffordAlgebra.equivProd : CliffordAlgebra (Q₁.prod Q₂) ≃ₐ[R] (evenOdd Q₁ ᵍ⊗[R] evenOdd Q₂)`
+
+## TODO:
+
+Introduce morphisms and equivalences of graded algebas, and upgrade `CliffordAlgebra.equivProd` to a
+graded algebra equivalence.
+
+-/
+
+suppress_compilation
+
+variable {R M₁ M₂ N : Type*}
+variable [CommRing R] [AddCommGroup M₁] [AddCommGroup M₂] [AddCommGroup N]
+variable [Module R M₁] [Module R M₂] [Module R N]
+
+variable (Q₁ : QuadraticForm R M₁) (Q₂ : QuadraticForm R M₂) (Qₙ : QuadraticForm R N)
+
+open scoped TensorProduct
+
+namespace CliffordAlgebra
+
+
+section map_mul_map
+
+variable {Q₁ Q₂ Qₙ}
+variable (f₁ : Q₁ →qᵢ Qₙ) (f₂ : Q₂ →qᵢ Qₙ) (hf : ∀ x y, Qₙ.IsOrtho (f₁ x) (f₂ y))
+variable (m₁ : CliffordAlgebra Q₁) (m₂ : CliffordAlgebra Q₂)
+
+/-- If `m₁` and `m₂` are both homogenous,
+and the quadratic spaces `Q₁` and `Q₂` map into
+orthogonal subspaces of `Qₙ` (for instance, when `Qₙ = Q₁.prod Q₂`),
+then the product of the embedding in `CliffordAlgebra Q` commutes up to a sign factor. -/
+nonrec theorem map_mul_map_of_isOrtho_of_mem_evenOdd
```

Comment:
Weirdly, after lifting things to `variable`, I now have to use `nonrec` here to avoid an error later on!